src/coreComponents/virtualElement/ConformingVirtualElementOrder1_impl.hpp

```diff
@@ -269,7 +269,11 @@ computeFaceIntegrals( arrayView2d< real64 const, nodes::REFERENCE_POSITION_USD >
   // Compute other geometrical properties.
   //  - compute rotation matrix.
   array2d< real64 > faceRotationMatrix( 3, 3 );
-  computationalGeometry::RotationMatrix_3D( faceNormal, faceRotationMatrix );
+  array1d< real64 > faceNormalArray( 3 );
+  faceNormalArray[0] = faceNormal[0];
+  faceNormalArray[1] = faceNormal[1];
+  faceNormalArray[2] = faceNormal[2];
```

Comment:
Can't these just be c-arrays?

Reply:
Thank you for the question! No, at the moment the method `computationalGeometry::RotationMatrix_3D` does not compile if the input normal is a c-array, and this is because [here](https://github.com/GEOSX/GEOSX/blob/310492d084cf66fb6cee730f5cd25e5be22eb03e/src/coreComponents/mesh/utilities/ComputationalGeometry.hpp#L307) and [here](https://github.com/GEOSX/GEOSX/blob/310492d084cf66fb6cee730f5cd25e5be22eb03e/src/coreComponents/mesh/utilities/ComputationalGeometry.hpp#L313) it calls [`genericTensorOps::crossProduct`](https://github.com/GEOSX/LvArray/blob/ab0d3025d2962aad10457b7bf7f5b6d266232276/src/genericTensorOps.hpp#L792) that does `internal::checkSizes<3>` at the beginning. This fails because it cannot determine the size of the c-array at compile time.

Reply:
it was suggested to me to tag @corbett5 here.

Reply:
That should work just fine, let me try it out.

Reply:
See #1432.

Also when you have a c-array as a function parameter it needs to be `real64 const (&cellCenter)[ 3 ]`, otherwise if it's just `real64 const cellCenter[3]` it's the same thing as `real64 const * cellCenter`.

Then finally when you run on the GPU you'll need to change the type of `faceRotatedVertices` from an `array2d` to a `stackArray2d< real64, 2 * MFN >` or something like that. You can't allocate an `array2d` on device.

Reply:
Many thanks for your help! I removed all `array1d` and `array2d` objects and declared c-array function parameters properly.